src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/MSITest.java

```diff
@@ -0,0 +1,189 @@
+import com.microsoft.sqlserver.jdbc.TestResource;
+import com.microsoft.sqlserver.jdbc.TestUtils;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.Constants;
+import com.microsoft.sqlserver.testframework.PrepUtil;
+
+
+/**
+ * Tests involving MSI authentication
+ */
+@RunWith(JUnitPlatform.class)
+@Tag(Constants.MSI)
+public class MSITest extends AESetup {
+    /*
+     * Test basic MSI auth
+     */
+    @Test
+    public void testAuth() throws SQLException {
+        try (SQLServerConnection con = PrepUtil.getConnection(connectionString)) {} catch (Exception e) {
+            fail(TestResource.getResource("R_loginFailed") + e.getMessage());
+        }
+    }
+
+    /*
+     * Test MSI auth using datasource
+     */
+    @Test
+    public void testDSAuth() throws SQLException {
+        SQLServerDataSource ds = new SQLServerDataSource();
+        AbstractTest.updateDataSource(connectionString, ds);
+
+        try (Connection con = ds.getConnection(); Statement stmt = con.createStatement()) {} catch (Exception e) {
+            fail(TestResource.getResource("R_loginFailed") + e.getMessage());
+        }
+    }
+
+    /*
+     * Test AKV with with credentials
+     */
+    @Test
+    public void testCharAkvWithCred() throws SQLException {
+        // unregister the custom providers registered in AESetup
+        SQLServerConnection.unregisterColumnEncryptionKeyStoreProviders();
+
+        // add credentials to connection string
+        String connStr = AETestConnectionString;
+        connStr = TestUtils.addOrOverrideProperty(connStr, Constants.KEYSTORE_AUTHENTICATION, "KeyVaultClientSecret");
+        connStr = TestUtils.addOrOverrideProperty(connStr, Constants.KEYSTORE_PRINCIPALID, keyStorePrincipalId);
+        connStr = TestUtils.addOrOverrideProperty(connStr, Constants.KEYSTORE_SECRET, keyStoreSecret);
+        testCharAkv(connStr);
+    }
+
+    /*
+     * Test AKV with MSI
+     */
+    @Test
+    public void testCharAkvWithMSI() throws SQLException {
+        // unregister the custom providers registered in AESetup
+        SQLServerConnection.unregisterColumnEncryptionKeyStoreProviders();
+
+        // set to use Managed Identity for keystore auth
+        String connStr = AETestConnectionString;
+        connStr = TestUtils.addOrOverrideProperty(connStr, Constants.KEYSTORE_AUTHENTICATION,
+                "KeyVaultManagedIdentity");
+        testCharAkv(connStr);
+    }
+
+    /*
+     * Test AKV with with bad credentials
+     */
+    @Test
+    public void testNumericAkvWithBadCred() throws SQLException {
+        // unregister the custom providers registered in AESetup
+        SQLServerConnection.unregisterColumnEncryptionKeyStoreProviders();
+
+        // add credentials to connection string
+        String connStr = AETestConnectionString;
+        connStr = TestUtils.addOrOverrideProperty(connStr, Constants.KEYSTORE_AUTHENTICATION, "KeyVaultClientSecret");

```

Comment:
I would prefer to use `KeyVaultManagedIdentity` here, that is what the feature is about.